force utf-8 for notebook json requests

bollwyvl opened the pull request:
JSON (and therefore all ipynb) should be served as utf-8.

This proposes that the url provider always use `utf-8` to decode what is coming back.

The default of [`ISO-8859-1`](https://github.com/jupyter/nbviewer/blob/4df8cfd919c87e03c029638b0b060555cd340c51/nbviewer/utils.py#L98) when `text/*` is received could stand, but maybe even it should be `utf-8`?

bollwyvl commented:
In response to #507 :)

damianavila commented:
Tests are failing but from a quick view I am not sure if it is related...

bollwyvl commented:
> Tests are failing but from a quick view I am not sure if it is related...

yep, just needed to rerun them....

I didn't add a test, or demonstrate a failing one... should probably do that...

damianavila commented:
> I didn't add a test, or demonstrate a failing one... should probably do that...

Yep

bollwyvl commented:
Sorry for the delay!

I've added a test case looking for `ñ` in `/urls/bitbucket.org/sandiego206/asdasd/raw/master/Untitled.ipynb`.

I've added a partial implementation of a simple use of tornado.testing, a la #423, but it is still going out to the network. Definitely thinking of how to do the real mocking, though!

bollwyvl commented:
Had to do some more refactoring of when options get initialized, parsed.

bollwyvl commented:
Ok, tests passing! Underlying question remains: is Latin-1 really what we want to default to in the absence of any other guidance, or should that be `utf-8`?

minrk commented:
@bollwyvl my thinking: since we know of at least one case of a server providing utf8 data without specifying, and we *don't* know of any such cases serving latin1, let's switch it to utf8 for now, and keep an eye out for encoding-related bug reports.

bollwyvl commented:
Changed the default to UTF-8... Travis gardening done... I think this is ready to go!

minrk commented:
Thanks!
